src/server/routes/actions.ts

```diff
@@ -23,43 +26,49 @@ router.get<unknown, BalanceResponse>("/balance", (_, res) => {
     });
 });
 
-const dripRequestHandler = async (requestOpts: DripRequestType): Promise<DripResponse> => {
-  const { address, parachain_id, amount, sender } = requestOpts;
-  metricsDefinition.data.total_requests++;
+const dripRequestHandler = new DripRequestHandler(actions, storage, recaptchaService);
 
-  const isAllowed = await storage.isValid(sender, address);
-  const isPrivileged = isAccountPrivileged(sender);
-  const isAccountOverBalanceCap = await actions.isAccountOverBalanceCap(address);
-
-  // parity member have unlimited access :)
-  if (!isAllowed && !isPrivileged) {
-    return { error: `${sender} has reached their daily quota. Only request once per day.` };
-  } else if (isAllowed && isAccountOverBalanceCap && !isPrivileged) {
-    return { error: `${sender}'s balance is over the faucet's balance cap` };
-  } else {
-    const sendTokensResult = await actions.sendTokens(address, parachain_id, amount);
-
-    // hash is null if something wrong happened
-    if (isDripSuccessResponse(sendTokensResult)) {
-      metricsDefinition.data.success_requests++;
-      storage.saveData(sender, address).catch((e) => {
-        logger.error(e);
-        errorCounter.plusOne("other");
-      });
+router.post<unknown, DripResponse, Partial<DripRequestType>>("/drip", async (req, res) => {
+  try {
+    const { address, parachain_id, amount, sender, recaptcha } = req.body;
+    if (!address) {
+      res.send({ error: "Missing parameter: 'address'" });
+      return;
     }
-
-    return sendTokensResult;
+    if (!parachain_id) {
```

Comment:
parachain_id is optional, so I guess we shouldn't return an error

Reply:
I didn't realize that `parachain_id` can be empty - updated.